securedrop/server.py

```diff
@@ -235,6 +256,23 @@ async def add_contact(self, addc, stream):
         msg = self.users.add_contact(self.sock_to_email[stream], addc.name, addc.email)
         await self.write_status(stream, msg)
 
+    async def list_contacts(self, stream):
+        # three verification steps
+        current_user_email = self.sock_to_email[stream]
+        # 1: contacts_dict contains the names and email adresses that a user has added
+        contacts_dict = self.users.list_contacts(current_user_email)
+        online_users_list = list(self.email_to_sock.keys())
+        contacts_dict_send = dict()
+
+        for email, name in contacts_dict.items():
+            other_user_contact_list = list(self.users.list_contacts(email))
```

Comment:
You also don't need this variable. You can check below with `current_user_email in self.users.list_contacts(email)`.

Reply:
Changed.